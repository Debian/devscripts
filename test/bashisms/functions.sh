#!/bin/sh

foo() {
    :
}

_all_good () {
    :
}

_all_good101_ ( ) {
    :
}

function BASHISM() {
    :
}

function BASHISM {
    :
}

function BASHISM {
    echo foo
} &>/dev/null # BASHISM

,() { # BASHISM
    :
}

function foo:bar:BASHISM { # BASHISMS
    :
}

function foo-bar-BASHISM() { # BASHISMS
    :
}

foo-bar-BASHISM ( ) {
    :
}

_ () {
    :
}

function _ { #BASHISM
    :
}

=() { #BASHISM
    :
}

function BASHISM=() { #BASHISMS
    :
}

function BASHISM= { #BASHISMS
    :
}

# This doesn't work:
#foo=() {
#    :
#}
