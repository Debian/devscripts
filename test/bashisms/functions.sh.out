possible bashism in bashisms/functions.sh line 15 ('function' is useless):
function BASHISM() {
possible bashism in bashisms/functions.sh line 19 ('function' is useless):
function BASHISM {
possible bashism in bashisms/functions.sh line 23 ('function' is useless):
function BASHISM {
possible bashism in bashisms/functions.sh line 25 (should be >word 2>&1):
} &>/dev/null # BASHISM
possible bashism in bashisms/functions.sh line 27 (function names should only contain [a-z0-9_]):
,() { # BASHISM
possible bashism in bashisms/functions.sh line 31 ('function' is useless):
function foo:bar:BASHISM { # BASHISMS
possible bashism in bashisms/functions.sh line 31 (function names should only contain [a-z0-9_]):
function foo:bar:BASHISM { # BASHISMS
possible bashism in bashisms/functions.sh line 35 ('function' is useless):
function foo-bar-BASHISM() { # BASHISMS
possible bashism in bashisms/functions.sh line 35 (function names should only contain [a-z0-9_]):
function foo-bar-BASHISM() { # BASHISMS
possible bashism in bashisms/functions.sh line 39 (function names should only contain [a-z0-9_]):
foo-bar-BASHISM ( ) {
possible bashism in bashisms/functions.sh line 47 ('function' is useless):
function _ { #BASHISM
possible bashism in bashisms/functions.sh line 51 (function names should only contain [a-z0-9_]):
=() { #BASHISM
possible bashism in bashisms/functions.sh line 55 ('function' is useless):
function BASHISM=() { #BASHISMS
possible bashism in bashisms/functions.sh line 55 (function names should only contain [a-z0-9_]):
function BASHISM=() { #BASHISMS
possible bashism in bashisms/functions.sh line 59 ('function' is useless):
function BASHISM= { #BASHISMS
possible bashism in bashisms/functions.sh line 59 (function names should only contain [a-z0-9_]):
function BASHISM= { #BASHISMS
